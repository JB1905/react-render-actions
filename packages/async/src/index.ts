export { Async } from './Async';
export { Await } from './Await';
export { Try } from './Try';
export { Catch } from './Catch';
export { Finally } from './Finally';
