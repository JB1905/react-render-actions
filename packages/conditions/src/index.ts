export { If } from './If';
export { Else } from './Else';
export { Then } from './Then';
export { When } from './When';
export { Unless } from './Unless';
