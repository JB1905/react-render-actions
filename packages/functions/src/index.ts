export { Map } from './Map';
export { Filter } from './Filter';
export { Reduce } from './Reduce';
