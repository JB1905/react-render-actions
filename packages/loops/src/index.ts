export { For } from './For';
export { ForOf } from './ForOf';
export { ForIn } from './ForIn';
export { ForEach } from './ForEach';
export { While } from './While';
export { DoWhile } from './DoWhile';
export { Break } from './Break';
export { Continue } from './Continue';
