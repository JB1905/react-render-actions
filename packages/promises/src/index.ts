export { RPromise } from './Promise';
export { Resolve } from './Resolve';
export { Reject } from './Reject';
