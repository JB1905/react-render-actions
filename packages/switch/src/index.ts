export { Switch } from './Switch';
export { Case } from './Case';
export { Default } from './Default';
