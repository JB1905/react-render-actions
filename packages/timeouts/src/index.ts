export { SetTimeout } from './SetTimeout';
export { SetInterval } from './SetInterval';
