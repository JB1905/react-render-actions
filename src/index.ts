export { For } from './Loops/For';
export { ForOf } from './Loops/ForOf';
export { ForIn } from './Loops/ForIn';
export { ForEach } from './Loops/ForEach';
export { While } from './Loops/While';
export { DoWhile } from './Loops/DoWhile';
export { Break } from './Loops/Break';
export { Continue } from './Loops/Continue';

export { SetTimeout } from './Timeouts/SetTimeout';
export { SetInterval } from './Timeouts/SetInterval';

export { Map } from './Functions/Map';
export { Filter } from './Functions/Filter';
export { Reduce } from './Functions/Reduce';
