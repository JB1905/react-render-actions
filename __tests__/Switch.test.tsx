import React from 'react';
import { render } from '@testing-library/react';

import { Switch, Case, Default } from '../src';

describe('Switch', () => {
  it.each([
    ['banana', 'Banana'],
    ['cherry', 'Fruit not found'],
  ])('should return $result for $switchValue', (switchValue, result) => {
    const { container, getByText } = render(
      <Switch switchValue={switchValue}>
        <Case case="apple">Apple</Case>
        <Case case="banana">Banana</Case>
        <Case case="pear">Pear</Case>
        <Case case="raspberry">Raspberry</Case>

        <Default>Fruit not found</Default>
      </Switch>
    );

    expect(container).toMatchSnapshot();

    expect(getByText(result)).toBeDefined();
  });

  it('should return neither case nor default', () => {
    const { container } = render(
      <Switch switchValue="pineapple">
        <Case case="apple">Apple</Case>
        <Case case="pear">Pear</Case>
      </Switch>
    );

    expect(container).toMatchSnapshot();
  });

  it('should return first case matching switchValue', () => {
    const { container } = render(
      <Switch switchValue="apple">
        <Case case="apple">Apple 1</Case>
        <Case case="pear">Pear</Case>
        <Case case="apple">Apple 2</Case>
        <Case case="apple">Apple 3</Case>
        <Case case="banana">Banana</Case>
      </Switch>
    );

    expect(container).toMatchSnapshot();
  });
});
