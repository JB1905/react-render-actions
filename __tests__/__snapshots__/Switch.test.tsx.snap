// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should return $result for $switchValue 1`] = `
<div>
  Banana
</div>
`;

exports[`Switch should return $result for $switchValue 2`] = `
<div>
  Fruit not found
</div>
`;

exports[`Switch should return first case matching switchValue 1`] = `
<div>
  Apple 1
</div>
`;

exports[`Switch should return neither case nor default 1`] = `<div />`;
